modules/alloydb/variables.tf

```diff
@@ -277,6 +277,15 @@ variable "project_id" {
   type        = string
 }
 
+variable "psc_config" {
```

Comment:
I think we can merge it with `network_config`, especially given the comment: `Network configuration for cluster and instance. Only one between cluster_network_config and cluster_psc_config can be used.`

If the above comment means either `network_config` or `psc_config` should be provided - we can have network_config of following type:
```hcl
type= object({
psa_config = optional(object({ <current network_config>}), null)
psc_config = optional(object({<what we need for psc_config}), null)
})
```

And add validation, that only one of the above is provided.